api/app/Models/User.php

```diff
@@ -1134,4 +1136,53 @@ public function getIsEmailVerifiedAttribute()
     {
         return $this->hasVerifiedEmail();
     }
+
+    public static function hydrateSnapshot(mixed $snapshot): Model|array
+    {
+        $user = new User();
+
+        $fields = [
+            'first_name' => ['firstName'],
+            'last_name' => ['lastName'],
+            'email' => ['email'],
+            'telephone' => ['telephone'],
+            'current_city' => ['currentCity'],
+            'current_province' => ['currentProvince', true],
+            'preferred_lang' => ['preferredLang', true],
+            'preferred_language_for_interview' => ['preferredLanguageForInterview', true],
+            'preferred_language_for_exam' => ['preferredLanguageForExam', true],
+            'citizenship' => ['citizenship', true],
+            'armed_forces_status' => ['armedForcesStatus', true],
+            'looking_for_english' => ['lookingForEnglish'],
+            'looking_for_french' => ['lookingForFrench'],
+            'looking_for_bilingual' => ['lookingForBilingual'],
+            'first_official_language' => ['firstOfficialLanguage', true],
+            'second_language_exam_completed' => ['secondLanguageExamCompleted'],
+            'second_language_exam_validity' => ['secondLanguageExamValidity'],
+            'comprehension_level' => ['comprehensionLevel', true],
+            'written_level' => ['comprehensionLevel', true],
+            'verbal_level' => ['verbalLevel', true],
+            'estimated_language_ability' => ['estimatedLanguageAbility', true],
+            'is_gov_employee' => ['isGovEmployee'],
+            'gov_employee_type' => ['govEmployeeType', true],
+            'has_priority_entitlement' => ['hasPriorityEntitelment'],
+            'priority_number' => ['priorityNumber'],
+            'location_preferences' => ['locationPreferneces', true],
+            'location_exemptions' => ['locationExcemptions'],
+            'accepted_operational_requirements' => ['acceptedOperationalRequirements', true],
```

Comment:
Oh, some typos

![image](https://github.com/user-attachments/assets/b9c84f99-99ba-4d79-bd19-dd96ce2471d0)

Reply:
oops, should be fixed in [fix typos](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11193/commits/3f2a1078ec38d01ac4bfeb6c608b82d9eb1792a1)